manifests/revoke.pp

```diff
@@ -25,10 +25,40 @@
 
   $etc_directory = $openvpn::etc_directory
 
-  exec { "revoke certificate for ${name} in context of ${server}":
-    command  => ". ./vars && ./revoke-full ${name}; echo \"exit $?\" | grep -qE '(error 23|exit (0|2))' && touch revoked/${name}",
-    cwd      => "${etc_directory}/openvpn/${server}/easy-rsa",
-    creates  => "${etc_directory}/openvpn/${server}/easy-rsa/revoked/${name}",
-    provider => 'shell',
+  case $openvpn::easyrsa_version {
+    '2.0': {
+      exec { "revoke certificate for ${name} in context of ${server}":
+        command  => ". ./vars && ./revoke-full ${name}; echo \"exit $?\" | grep -qE '(error 23|exit (0|2))' && touch revoked/${name}",
+        cwd      => "${etc_directory}/openvpn/${server}/easy-rsa",
+        creates  => "${etc_directory}/openvpn/${server}/easy-rsa/revoked/${name}",
+        provider => 'shell',
+      }
+    }
+    '3.0': {
+      if $openvpn::manage_service {
+        if $facts['service_provider'] == 'systemd' {
+          $lnotify = Service["openvpn@${server}"]
+        } elsif $openvpn::namespecific_rclink {
+          $lnotify = Service["openvpn_${server}"]
+        } else {
+          $lnotify = Service['openvpn']
+          Openvpn::Server[$server] -> Service['openvpn']
+        }
+      }
+      else {
+        $lnotify = undef
+      }
+
+      exec { "revoke certificate for ${name} in context of ${server}":
+        command  => ". ./vars && echo yes | ./easyrsa revoke ${name} 2>&1 | grep -E 'Already revoked|was successful|not a valid certificate' && ./easyrsa gen-crl && /bin/cp -f keys/crl.pem ../crl.pem && touch revoked/${name}",
```

Comment:
May I ask, why you are using 'echo yes' instead of the --batch switch on easyrsa?

I just hacked together a 'revoke-full' temp fix and found this to be simpler.
`./easyrsa --batch revoke ${name}`

Cheers,
Jan

Reply:
No reason, just missed that flag.